Seneca-CDOT/telescope
# Add tool script to run backup

## Issue This PR Addresses

Fixes #2822 

## Type of Change

- [X] **Development Tools**: Change which improves development tools

## Description

This PR adds a small Docker container that will run a backup script as a cron job, it also bundles up another script to run a restoration using one of the available database dumps.

The scripts in question are:

- `create-backup.sh`, which creates a dump file by using `pg_dump`. The dump file will list all data definitions and data values from all tables in a specific database).
- `restore-backup.sh`, which will use a specific dump file to restore a database with `pg_restore`. You are supposed to use the dump file generated by the `create-backup` file.

## Steps to test the PR

1. Start the services.
2. Enter some information into a few tables.
3. Login into the `pg-backup-cron-job` container (`docker exec -it pg-backup-cron-job sh`).
4. Instead of waiting for the cron job to be executed, you may execute the file with the following command: `/etc/periodic/daily/create-backup`. After the command has been executed, there should be a file in `/var/opt/`, similar to `2022-04-08-02231.dump`, make a note of its name.
5. After getting the backup file, modify the database by deleting those tables with data.
6. To restore those tables, run the `restore-backup` file with the following command: `./restore-backup /var/opt/2022-04-08-02231.dump`, for example.

## Checklist
